plugins/modules/notification/mattermost.py

```diff
@@ -99,12 +123,16 @@ def main():
         argument_spec=dict(
             url=dict(type='str', required=True),
             api_key=dict(type='str', required=True, no_log=True),
-            text=dict(type='str', required=True),
+            text=dict(type='str', default=None),
```

Comment:
`None` is always the default, there's no need to specify it.
```suggestion
            text=dict(type='str'),
```